### DISCUSSION TITLE
Errors: The default export is not a React Component in page: "/_error" and TypeError: __webpack_require__.C is not a function

### DISCUSSION THREAD

**Discussion by nick-myers-dt** [vercel/next.js#31872] [Help]
### Summary

I have a small Next.js v12.0.4 project which starts reporting these errors when developing locally with `next dev`:

- `The default export is not a React Component in page: "/_error"`
- `error - TypeError: __webpack_require__.C is not a function`
 
### How the issue manifests`

1. `next dev`
2. Everything works for a few minutes
3. Refresh `localhost:3000`
4. Observe error (full error below)
5. Kill process and start again

I am not using a `next.config.js`file

I've spent days trying to work out why this happens, or how to reproduce it on a clean repo for a bug report, and cannot, so am asking the community for help.

At the bottom of this post, I've included the only reference to the Next.js `<ErrorPage />` made in the `index.tsx` file.

Any advice would be very welcome at this point, before I trash the whole project and start over.

### Full error

```bash
❯ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /Users/nickmyers/repos/rebuild/project/.env.local
event - compiled successfully in 3s (189 modules)
wait  - compiling /...
event - compiled successfully in 1511 ms (879 modules)
wait  - compiling /api/building/[number] (server only)...
event - compiled successfully in 220 ms (893 modules)
wait  - compiling...
event - compiled successfully in 759 ms (869 modules)
wait  - compiling...
event - compiled client successfully in 268 ms (842 modules)
wait  - compiling...
event - compiled client successfully in 355 ms (842 modules)
wait  - compiling...
event - compiled client successfully in 402 ms (842 modules)
wait  - compiling /api/register/[number] (server only)...
event - compiled successfully in 177 ms (883 modules)
wait  - compiling /...
event - compiled successfully in 340 ms (892 modules)
wait  - compiling /_error...
event - compiled successfully in 225 ms (893 modules)
Error: The default export is not a React Component in page: "/_error"
    at Object.renderToHTML (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/render.js:248:19)
    at doRender (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/next-server.js:1428:57)
    at /Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/next-server.js:1523:34
    at /Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/response-cache.js:63:42
    at ResponseCache.get (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/response-cache.js:80:11)
    at DevServer.renderToResponseWithComponents (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/next-server.js:1459:53)
    at DevServer.renderErrorToResponse (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/next-server.js:1709:35)
    at async DevServer.renderToResponse (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/next-server.js:1641:30)
    at async DevServer.pipe (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/next-server.js:1148:25)
    at async Object.fn (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/next-server.js:949:21)
error - TypeError: __webpack_require__.C is not a function
```

### Versions

Node: v14.18.1
Next: v12.0.4
React: v17.0.2

### Project files (excluding .next, .yarn, node_modules)
```bash
project/_app.tsx
project/index.tsx
project/api/building/[number].tsx
project/api/register/[number].tsx
project/lib/aws.ts (gets secrets from AWS)
project/lib/okta.ts (creates users in Okta)
project/lib/organisation.ts (SQL queries to database)
project/public/logo.png
project/public/favicon.ico
project/.env.local
project/.gitignore
project/.eslintrc.json
project/.prettierrc.json
project/.yarnrc.yml
project./next-env.d.ts
project/package.json
project/tsconfig.json
project/yarn.lock
```

### Package.json

```json
{
  "name": "project",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "prettier": "onchange '**/*' -- prettier --write '{{changed}}'"
  },
  "next-unused": {
    "alias": {},
    "include": [
      "pages"
    ],
    "exclude": [],
    "entrypoints": [
      "pages"
    ]
  },
  "dependencies": {
    "aws-sdk": "^2.1035.0",
    "formik": "^2.2.9",
    "formik-semantic-ui-react": "^2.0.0",
    "isomorphic-fetch": "^3.0.0",
    "next": "12.0.4",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-google-recaptcha-v3": "^1.9.7",
    "semantic-ui-css": "^2.4.1",
    "semantic-ui-react": "^2.0.4",
    "sqlstring": "^2.3.2",
    "validator": "^13.7.0"
  },
  "devDependencies": {
    "@types/node": "16.11.9",
    "@types/react": "17.0.35",
    "@types/sqlstring": "^2.3.0",
    "@types/validator": "^13.7.0",
    "@typescript-eslint/eslint-plugin": "^5.4.0",
    "@typescript-eslint/parser": "^5.4.0",
    "eslint": "^7.32.0",
    "eslint-config-next": "12.0.4",
    "eslint-config-prettier": "^8.3.0",
    "onchange": "^7.1.0",
    "prettier": "^2.4.1",
    "typescript": "4.3",
    "webpack": "^5.64.4"
  },
  "packageManager": "yarn@3.1.1"
}
```
Note: I don't use webpack directly, I only installed it as another thread suggested it might resolve the `TypeError` issue.

### tsconfig.json

```json
{
  "compilerOptions": {
    "target": "ES6",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules", ".next"]
}
```

### _app.tsx

```tsx
import 'semantic-ui-css/semantic.min.css'
import type { AppProps } from 'next/app'

function MyApp({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />
}

export default MyApp
```

### index.tsx

```tsx

import { GoogleReCaptchaProvider, useGoogleReCaptcha } from 'react-google-recaptcha-v3'
import { getAwsSecretAsync } from '../lib/aws'
import Head from 'next/head'
import { GetServerSideProps, NextPage } from 'next'
import ErrorPage from 'next/error'

const Project: NextPage<{ captchaPublicKey: string }> = (props) => {
  if (!props.captchaPublicKey) {
    return <ErrorPage statusCode={500} />
  }
  return (
    <>
      <Head>
        <title>Project Title</title>
      </Head>
      <GoogleReCaptchaProvider reCaptchaKey={props.captchaPublicKey}>
        <Heading />
      ....
      </GoogleReCaptchaProvider>
    </>
  )
}

export default Project

export const getServerSideProps: GetServerSideProps = async ({ res }) => {
  try {
    let captchaPublicKey = ''
    if (process.env.NEXT_PUBLIC_RECAPTCHA_SITE_KEY) {
      console.log('running locally')
      captchaPublicKey = process.env.NEXT_PUBLIC_RECAPTCHA_SITE_KEY
    } else {
      console.log('running hosted')
      const reCaptchaKeySecret = await getAwsSecretAsync('/lib/google/recaptcha/publicSiteKey')
      captchaPublicKey = reCaptchaKeySecret as string
    }
    console.log(captchaPublicKey)

    return {
      props: {
        captchaPublicKey: captchaPublicKey,
      },
    }
  } catch (error) {
    res.statusCode = 500
    return {
      props: {},
    }
  }
}
```

**Comment by shavidze** [comment]
I faced this kind of error a while ago. I just resolved this issue by exporting my component at the bottom of the page. I am not 100% sure if it will work for you but let's try

**Reply by nick-myers-dt** [reply]
Thank you @shavidze I did try this (and did so again just now) however it then has the effect of my `index.tsx` page rendering a blank page.  

If I restart the `next dev` process however, I do get an error:

```bash
TypeError: __webpack_require__.C is not a function
    at /Users/nickmyers/repos/rebuild/project/.next/server/pages/_document.js:151:21
    at Object.<anonymous> (/Users/nickmyers/repos/rebuild/project/.next/server/pages/_document.js:156:3)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.requirePage (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/require.js:48:12)
    at Object.loadComponents (/Users/nickmyers/repos/rebuild/project/node_modules/next/dist/server/load-components.js:56:23)
error - SyntaxError: /Users/nickmyers/repos/rebuild/project/.next/react-loadable-manifest.json: Unexpected end of JSON input
```

I can confirm that `react-loadable-manifest.json` is an empty file

**Reply by nick-myers-dt** [reply]
I've just deleted `.next` and run `next dev` again, to see if the same issue arises.

**Reply by nick-myers-dt** [reply]
I can confirm that after deleting `.next` and running `next dev` initially the page renders, however after a few seconds, both `build-manifest.json` and `react-loadable-manifest.json` are updated to be empty files and the page is not rendering anything (no errors). 

I'm entirely confused.

**Reply by sneko** [reply]
@nick-myers-dt did it happen to you again after that?

Facing this weird behavior... `next dev` works for seconds, and after I get this :/ it makes absolutely no sense. I deleted the `node_modules` `.cache``.next``.turbo`... but still an issue. Haha it's a real pain ^^

**Reply by nick-myers-dt** [reply]
@sneko I'm sorry, this was so long ago now that I don't recall specifically how I resolved it.

**Reply by Chacho104** [reply]
Just realised that when using route grouping, this error occurs when you have an empty layout file within your route group. To resolve it, write code for layout file or delete it for the time being.

**Reply by aviroopjana** [reply]
Thanks for this info

**Reply by binrusdi** [reply]
@Chacho104 yap, is true. i delete a file layout.js not field.

thankyou brother

**Reply by aravind-allamneni** [reply]
This helped. I was getting the same error because of the empty layout file

**Reply by KENG85** [reply]
Deleted the file and I'm good.  Thank you!

**Reply by Raj067** [reply]
Works for me too, thanks

**Reply by shaeebali** [reply]
Thanks, same here. Deleted the "layout" file and it works a charm.

**Reply by tim-anderson-linksbridge** [reply]
just an update:  I changed the layout to const layout = () => {} then export default layout too, then it worked.  Make sure to do the export default with both page and layout.tsx

**Reply by Stminty** [reply]
Thank you thank you thank you! Deleted layout and its sorted.

**Comment by kennedy-muthaura** [comment]
This worked for me
1. Delete node_modules folder and yarn.lock or package-lock.json
2. run yarn install or npm install

**Reply by de-mawo** [reply]
Thanks @kennedy-muthaura , this worked for me

**Reply by kennedy-muthaura** [reply]
welcome @de-mawo , glad to have helped

**Reply by TahminaTania** [reply]
worked for me as well .. thank you

**Reply by lokichaulagain** [reply]
Thanks worked for me .

**Comment by FriedrichMatheus** [comment]
in your package.json you need the dependency "@types/react-dom" for the inline export to work

**Comment by RanaReebaal** [comment]
I resolved this error by erasing all the code in my layout.tsx file and then restoring it to its default code ( You can do it with the help of Ctrl+Z ). Maybe it would also help you...... :)

**Reply by yevhen-logosha** [reply]
this set me on the right track, I was getting 
```
default export is not a React Component in page: "undefined"
```
What it was in the end - in my `layout.tsx` I was doing `export const Layout = ....`
What I should have (and did in the end to solve this) was make my Layout a `default` export.

**Reply by akins-dev** [reply]
Thanks, this helped me realize that I didn't export the `layout.tsx`

**Reply by esbnet** [reply]
It is necessary to export Layout as default. Thanks for the tip!

**Comment by Eshwar1212-maker** [comment]
If anyone is getting this error whil trying to dynamically intercept routes, writing this code fixed it for me under the dynamic route you want to intercept(for example (@authModal): 

import type { FC } from "react";

const Default: FC = () => {
  return null;
};

export default Default;

instead of: 

export function Default(){
    return null
} 
which was not working

**Reply by YangDaniu** [reply]
it helps me a lot

**Comment by Gautamvaishnav-git** [comment]
In my case, I just exit the server and start it again.

**Comment by cryptoHersh** [comment]
This most likely happen if you are using app folder and appRouter, and you have empty layout.tsx file in sub route. Remove layout.tsx  and try again.

**Reply by AKM1189** [reply]
It saves me from spending more time finding the error.

**Comment by Stephan-MC** [comment]
I am on next 15 and still getting this issue where all was working fine. I tried added another adjacent loadint.tsx file for some other route and bumped into this. The react ecosystem is real CRAP.
